tokio-util/src/codec/length_delimited.rs

```diff
@@ -1018,6 +1037,39 @@ impl Builder {
         self.num_skip
             .unwrap_or(self.length_field_offset + self.length_field_len)
     }
+
+    fn adjust_max_frame_len(&mut self, adjust_to_max: bool) {
+        // This function is basically `std::usize::saturating_add_signed`. Since it
+        // requires MSRV 1.66, its implementation is copied here.
+        //
+        // TODO: use the method from std when MSRV becomes >= 1.66
+        fn saturating_add_signed(num: usize, rhs: isize) -> usize {
+            let (res, overflow) = num.overflowing_add(rhs as usize);
+            if overflow == (rhs < 0) {
+                res
+            } else if overflow {
+                usize::MAX
+            } else {
+                0
+            }
+        }
+
+        // Calculate the maximum number that can be represented using `length_field_len` bytes.
+        let max_number = 2usize.saturating_pow((8 * self.length_field_len - 1) as u32);
+        // In order to prevent an overflow, we do the last part manually.
+        let max_number = max_number + (max_number - 1);
```

Comment:
This looks like it can overflow.

Reply:
It did before (on BSD 32bit) and I added this test `max_allowed_frame_fits` to catch it. 

On a 32bit system it can still overflow if `length_field_len` is more than 4 bytes. But I thought this is an invalid value anyway(?).

Reply:
The docs say:

> The default value is 4. The max value is 8.